packages/catalog-process/src/services/catalogService.ts

```diff
@@ -1824,9 +1824,7 @@ export function catalogServiceBuilder(
 
       if (
         eservice.data.descriptors.every(
-          (descriptor) =>
-            descriptor.state === descriptorState.draft ||
-            descriptor.state === descriptorState.archived
+          (descriptor) => descriptor.state === descriptorState.draft
```

Comment:
Is it possible to update the name of an archived eservice? If not, I would keep this check... even if it's a case that cannot be triggered by the current UI/flow, the back-end business logic should defend against all edge cases that are allowed by the data model.
Here I would reuse `isDescriptorUpdatable`, it's the exhaustive check we use in other update methods that can be called after the e-service is published (same comment for https://github.com/pagopa/interop-be-monorepo/pull/1384)